numalogic/udfs/factory.py

```diff
@@ -27,6 +26,8 @@ class UDFFactory:
     from numalogic.udfs.preprocess import PreprocessUDF
     from numalogic.udfs.trainer import DruidTrainerUDF, PromTrainerUDF
 
+    nl_udf_t = TypeVar("nl_udf_t", bound=NumalogicUDF, covariant=True)
```

Comment:
Covariant =True not needed. It should be strictly bounded to NumalogicUDF.

Reply:
Not really. For example PostprocessUDF has multi compute functions. We need the linter to recognize the child classes of NumalogicUDF